### PR TITLE
Revert formSubmit, add isBrowser check

### PR DESCRIPTION
**Pull request by jordanranz**
_Issue #, if available:_
https://github.com/aws-amplify/amplify-js/issues/5325

_Description of changes:_
- Revert back to simulate form submit with button
- Add browser check for server side rendering

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-ui-components/src/components/amplify-button/amplify-button.tsx [hunk 1]
```diff
@@ -32,9 +33,15 @@ export class AmplifyButton {
         const formSection = this.el.closest('amplify-form-section');
         form = formSection && formSection.shadowRoot.querySelector('form');
       }
-      if (form) {
+      if (form && browserOrNode().isBrowser) {
```
Comment (ericclemmons) [inline packages/amplify-ui-components/src/components/amplify-button/amplify-button.tsx:36]: What error happens here?  It’s interesting that an event handler would run during SSR.

File: packages/amplify-ui-components/src/components/amplify-button/amplify-button.tsx [hunk 2]
```diff
@@ -32,9 +33,15 @@ export class AmplifyButton {
...
         ev.preventDefault();
-        form.requestSubmit();
+
+        const fakeButton = document.createElement('button');
+        fakeButton.type = this.type;
+        fakeButton.style.display = 'none';
+        form.appendChild(fakeButton);
+        fakeButton.click();
```
Comment (ericclemmons) [inline packages/amplify-ui-components/src/components/amplify-button/amplify-button.tsx:43]: I’m ignorant on Shadow DOM details. If the sign in button was `<button type=submit>`, would an onSubmit handler on form work?

I assume the shadow DOM prevents that from bubbling?

File: packages/amplify-ui-components/src/components/amplify-button/amplify-button.tsx (hunk 1 above)
Comment (jordanranz) [reply packages/amplify-ui-components/src/components/amplify-button/amplify-button.tsx]: got rid of this :D

File: packages/amplify-ui-components/src/components/amplify-button/amplify-button.tsx (hunk 2 above)
Comment (jordanranz) [reply packages/amplify-ui-components/src/components/amplify-button/amplify-button.tsx:42]: Spoke offline about this.

**Pull request merged**